Cython/Utility/Exceptions.c

```diff
@@ -907,12 +907,73 @@ static void __Pyx_AddTraceback(const char *funcname, int c_line,
 #if CYTHON_COMPILING_IN_LIMITED_API
 static void __Pyx_AddTraceback(const char *funcname, int c_line,
                                int py_line, const char *filename) {
+    PyObject *code_object = NULL, *py_py_line = NULL, *py_funcname = NULL, *dict = NULL;
+    PyObject *replace = NULL, *getframe = NULL, *frame = NULL;
+    PyObject *exc_type, *exc_value, *exc_traceback;
+    int success = 0;
     if (c_line) {
         // Avoid "unused" warning as long as we don't use this.
         (void) $cfilenm_cname;
         (void) __Pyx_CLineForTraceback(__Pyx_PyThreadState_Current, c_line);
     }
-    _PyTraceback_Add(funcname, filename, py_line);
+
+    // DW - this is a horrendous hack, but I'm quite proud of it. Essentially
```

Comment:
I'll just ignore this part in my review and believe both properties as you state them here.

Reply:
Yeah - it'd noticed that it could be cached, but was just trying to keep things simple for the initial version.

We can definitely come back and optimize this later.

I'm also open to different approaches, but I honestly couldn't think of anything else that might work.